orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/entity/VtrList.java

```diff
@@ -0,0 +1,159 @@
+        if (vtr != null && !vtr.isEmpty()) {
+            JSONParser parser = new JSONParser(DEFAULT_PERMISSIVE_MODE);
+            JSONArray vtrJsonArray;
+            try {
+                if (vtr.size() != 1) {
+                    throw new IllegalArgumentException("Expected VTR to have single entry.");
+                }
+
+                // The nimbus package parses the vtr from the auth request as a list with one
+                // element
+                // which is a json list
+                // The .get(0) below returns the json list - it doesn't just get the first element
+                // of
+                // the Vector of Trust
+                vtrJsonArray = (JSONArray) parser.parse(vtr.get(0));
+                if (!vtrJsonArray.isEmpty()) {
+                    return new VtrList(
+                            vtrJsonArray.stream()
+                                    .map(String.class::cast)
+                                    .map(VectorOfTrust::parse)
+                                    .toList());
+                }
+            } catch (net.minidev.json.parser.ParseException
+                    | ClassCastException
+                    | IllegalArgumentException e) {
+                LOG.error("Error when parsing VTR attribute.", e);
+                throw new IllegalArgumentException("Invalid VTR attribute.", e);
+            }
+        }
+
+        LOG.info("VTR attribute is not present. A default value will be used.");
+        return DEFAULT_VTR_LIST;
+    }
+
+    private static void validateVtr(List<VectorOfTrust> vtr) {
+        for (var vot : vtr) {
+            validateVot(vot);
+        }
+    }
+
+    private static void validateVot(VectorOfTrust vot) {
+        if (vot.getLevelOfConfidence() != LevelOfConfidence.NONE
+                && vot.getCredentialTrustLevel() == CredentialTrustLevel.LOW_LEVEL) {
+            throw new IllegalArgumentException(
+                    format(
+                            "Unsupported combination of \"Credential Trust Level\": \"{0}\" and \"Level of Confidence\": \"{1}\".",
+                            vot.getCredentialTrustLevel(), vot.getLevelOfConfidence()));
+        }
+
+        if (!vot.getLevelOfConfidence().isSupported()) {
+            throw new IllegalArgumentException(
+                    format(
+                            "Unsupported \"Level of Confidence\": \"{0}\".",
+                            vot.getLevelOfConfidence()));
+        }
+    }
+
+    public List<VectorOfTrust> getVtr() {
+        return vtr;
+    }
+
+    public CredentialTrustLevel getCredentialTrustLevel() {
+        return credentialTrustLevel;
+    }
+
+    public List<LevelOfConfidence> getLevelsOfConfidence() {
+        return levelsOfConfidence;
+    }
+
+    public VectorOfTrust getEffectiveVectorOfTrust() {
+        return effectiveVectorOfTrust;
+    }
+
+    public boolean identityRequired() {
+        return levelsOfConfidence.get(0) != LevelOfConfidence.NONE;
```

Comment:
Understand that the null check is no longer necessary, but I don't think we should lose the comment

Reply:
Comment added.

Reply:
Sorry this is super picky, but would you be able to restore the comment to how it was before? If in the future we do change this assumption in any way, it would be really helpful to have a consistent way to go back and look at everywhere it was used

Reply:
Sounds good to me 🙂

Reply:
Done